src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -758,6 +760,126 @@ private bool HasValidPlatformVersions()
             }
         }
 
+        internal static void AnalyzePruningResults(PackageSpec project, ILogger logger)
+        {
+            bool enablePruningWarnings =
+                SdkAnalysisLevelMinimums.IsEnabled(
+                    project.RestoreMetadata.SdkAnalysisLevel,
+                    project.RestoreMetadata.UsingMicrosoftNETSdk,
+                    SdkAnalysisLevelMinimums.PruningWarnings) &&
+                HasFrameworkNewerThanNET10(project);
+
+            if (!enablePruningWarnings)
+            {
+                return;
+            }
+
+            Dictionary<string, List<string>> prunedDirectPackages = null;
+
+            // Calculate direct packages that are in the pruning range.
+            foreach (TargetFrameworkInformation framework in project.TargetFrameworks)
```

Comment:
nitpick, if you put this loop in a method, and also the body of the following if statement, then this method becomes:

```csharp
bool enablePruningWarnings = ...;
if (!enablePruningWarnings)
{
    return;
}

Dictionary<string, List<string>> prunedDirectPackages = GetPrunedDirectPackages(...);

if (prunedDirectPackages != null)
{
    OutputWarnings(prunedDirectPackages);
}
```

and each method that gets called is small enough to fit on even a low resolution screen.